app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -0,0 +1,967 @@
+        ),
+      }
+    }
+
+    if (row.type === DiffRowType.Deleted) {
+      return {
+        ...row,
+        data: this.getRowDataPopulated(
+          row.data,
+          numRow,
+          DiffColumn.Before,
+          this.state.beforeTokens
+        ),
+      }
+    }
+
+    if (row.type === DiffRowType.Modified) {
+      return {
+        ...row,
+        beforeData: this.getRowDataPopulated(
+          row.beforeData,
+          numRow,
+          DiffColumn.Before,
+          this.state.beforeTokens
+        ),
+        afterData: this.getRowDataPopulated(
+          row.afterData,
+          numRow,
+          DiffColumn.After,
+          this.state.afterTokens
+        ),
+      }
+    }
+
+    if (row.type === DiffRowType.Context) {
+      const lineTokens =
+        getTokens(row.beforeLineNumber, this.state.beforeTokens) ??
+        getTokens(row.afterLineNumber, this.state.afterTokens)
+      const tokens = lineTokens ? [...row.tokens, lineTokens] : row.tokens
+
+      return {
+        ...row,
+        tokens,
+      }
+    }
+
+    return row
+  }
+
+  private getRowDataPopulated(
+    data: SimplifiedDiffRowData,
+    row: number,
+    column: DiffColumn,
+    tokens: ITokens | undefined
+  ): IDiffRowData {
+    const lineTokens = getTokens(data.lineNumber, tokens)
+
+    return {
+      ...data,
+      tokens: lineTokens ? [...data.tokens, lineTokens] : data.tokens,
+      isSelected: isInSelection(
+        data.diffLineNumber,
+        row,
+        column,
+        this.getSelection(),
+        this.state.temporarySelection
+      ),
+    }
+  }
+
+  private getDiffLineNumber(
+    rowNumber: number,
+    column: DiffColumn
+  ): number | null {
+    const rows = getDiffRows(this.props.diff, this.props.showSideBySideDiff)
+
+    if (rows[rowNumber] === undefined) {
+      return null
+    }
+
+    const row = rows[rowNumber]
+
+    if (row.type === DiffRowType.Added || row.type === DiffRowType.Deleted) {
+      return row.data.diffLineNumber
+    }
+
+    if (row.type === DiffRowType.Modified) {
+      return column === DiffColumn.After
+        ? row.afterData.diffLineNumber
+        : row.beforeData.diffLineNumber
+    }
+
+    return null
+  }
+
+  /**
+   * This handler is used to limit text selection to a single column.
+   * To do so, we store the last column where the user clicked and use
+   * that information to add a CSS class on the container div
+   * (e.g `selecting-before`).
+   *
+   * Then, via CSS we can disable text selection on the column that is
+   * not being selected.
+   */
+  private onMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (!this.props.showSideBySideDiff) {
+      return
+    }
+
+    // We need to use the event target since the current target will
+    // always point to the container.
+    const target = event.target as HTMLDivElement
+
+    const isSelectingBeforeText = target.closest('.before')
+    const isSelectingAfterText = target.closest('.after')
```

Comment:
This makes me a little uneasy. We don't know that the thing the  user happened to click on is a `div` do we? And it doesn't seem like we care either, we just want the `.closest` function. Could we do something like this instead?

```suggestion
    if (!(event.target instanceof HTMLElement)) {
      return
    }

    // We need to use the event target since the current target will
    // always point to the container.
    const isSelectingBeforeText = event.target.closest('.before')
    const isSelectingAfterText = event.target.closest('.after')
```